runtime/near-vm-runner/src/cache.rs

```diff
@@ -85,6 +87,110 @@ impl fmt::Debug for MockCompiledContractCache {
     }
 }
 
+struct FilesystemCompiledContractCacheState {
+    dir: rustix::fd::OwnedFd,
+    test_temp_dir: Option<tempfile::TempDir>,
+}
+
+#[derive(Clone)]
+pub struct FilesystemCompiledContractCache {
+    state: Arc<FilesystemCompiledContractCacheState>,
+}
+
+impl FilesystemCompiledContractCache {
+    pub fn new<HD: AsRef<std::path::Path> + ?Sized, SP: AsRef<std::path::Path> + ?Sized>(
+        home_dir: &HD,
+        store_path: Option<&SP>,
```

Comment:
Why do you need the fancy types HD and SP? Can you just use Path or String?

Reply:
Definitely not a string -- paths ain't necessarily utf-8. Path I could use here, but that makes for a particularly inconvenient interface if you need to pass literals in. home_dir could most likely be a regular `Path` since nothing is gonna bother passing a literal for it, but store path is less of a slam dunk, especially given use of this in tests.